### PR TITLE
Use redux for state management

### PR DESCRIPTION
**Pull request by kristinhortsch**

### CODE REVIEW

**Review by grayson073 — CHANGES_REQUESTED** [review]
Noticed some redundancy between actions/reducers (your two reducer cases return the same thing). Also, only one is being utilized in the application while the other exists only in the test file. Please resolve and re-submit for review :)

**Comment by mehtaphysical** [comment]
Add some tests for reducers and selectors

**Comment by mehtaphysical** [comment]
Should this merge into master? Or dev?

File: src/reducers/markdownReducers.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { NEW_MARKDOWN_UPDATE } from '../actions/markdownActions';
+
+const initialState = {
+  markdown: ''
+};
+
+export default function reducer(state = initialState, action) {
+  switch(action.type) {
+    case NEW_MARKDOWN_UPDATE:
```
Comment (mehtaphysical) [inline src/reducers/markdownReducers.js]: This should be `UPDATE_MARKDOWN`

File: src/selectors/markdownSelectors.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const getAllMarkdowns = state => {
```
Comment (mehtaphysical) [inline src/selectors/markdownSelectors.js]: there is only one markdown file. should probably be `getMarkdown`

Comment (kristinhortsch) [reply src/selectors/markdownSelectors.js]: Agree on all accounts. We changed the PR to merge to dev. And fixed the other pieces!

File: src/components/markdown/Document.js [hunk 3]
```diff
@@ -2,24 +2,46 @@ import React, { PureComponent } from 'react';
...
   };
 
   updateMarkdown = ({ target }) => {
-    this.setState({ markdown: target.value });
+    store.dispatch(updateNewMarkdown(target.value));
   };
 
+  updateState = () => {
+    const currentReduxState = store.getState();
+    const markdown = getAllMarkdowns(currentReduxState);
```
Comment (mehtaphysical) [inline src/components/markdown/Document.js]: Need to change this. Don't forget to run your code to make sure it works :)

**Review by mehtaphysical — APPROVED** [review]

**Pull request merged**